### ISSUE TITLE
add integration tests for updateEconomicEvent

### ISSUE THREAD

**Issue by Connoropolous** [holo-rea/holo-rea#279]

**Comment by fosterlynn** [comment]
@Connoropolous events can't be deleted, see this section https://lab.allmende.io/valueflows/valueflows/-/wikis/Application-specification-guidance#validations which also has info about what can be updated.  

(That's an accounting thing that still applies to us, because you never know when a financial statement or other report was pulled.  I need to write up some guidance about "backing out" economic events, and have done a little research, but haven't written it yet.  But you shouldn't need that for what you are doing, that's a user thing.)

**Comment by Connoropolous** [comment]
there is a method in the vf-graphql though?

**Comment by fosterlynn** [comment]
I see what you mean, and there is a delete for EconomicResource too, which is also not deleteable.  @pospi any recollections on this?  Too far back for my brain....

**Comment by Connoropolous** [comment]
it is even implemented in the code already

**Comment by fosterlynn** [comment]
anyway, I wouldn't write a test for it, until we see if I am missing something or not

it occurs to me maybe there could be something around events that never are countersigned or otherwise accepted by the other party, so they never became actual events??  but on the basic functional level, you can't delete events or resources, if it is possible a financial report has been run including them

**Comment by pospi** [comment]
Yeah it might be an implementation / model logic type distinction, but I put those there for use in removing or fixing the occasional event that is logged purely erroneously & hasn't had anything else done to/with it since being created. For this to be a feature you would also need the capability to delete similar resources having been instantiated by any such events.

`UpdateEconomicResource` is probably mostly going away with some of the new `Action` verbs, but let's leave it there for now and follow through the updates naturally to be sure it's fully redundant.

Also deletion checks aren't implemented yet so yeah, this is confusing and likely vandalous API endpoints to use at this point (;

**Comment by fosterlynn** [comment]
>I put those there for use in removing or fixing the occasional event that is logged purely erroneously & hasn't had anything else done to/with it since being created.

That would be great, but there's no way to know if in the last 3 seconds (or 3 nano-seconds), someone pulled an official financial or tax report.

[edit] To be more clear, these reports are mostly date/time range driven, so deleting something in the last accounting period will mess things up.  It's an existing accounting practice, that keeps you clean in an audit.

**Comment by fosterlynn** [comment]
>UpdateEconomicResource is probably mostly going away with some of the new Action verbs, but let's leave it there for now and follow through the updates naturally to be sure it's fully redundant.

Unfortunately, it won't fully go away with the new action effects.  We could make it go away, but it still seems confusing and excessive to make people log a new event when they are just fixing a typo or clarifying a description.  We tried to split off economic effects and other stuff.

**Comment by Connoropolous** [comment]
okay this is getting interesting. Sounds like a pretty grey area. 
I still think that it could be useful to add the integration tests, which wouldn't be a lot of work, because otherwise we code that technically is written, but could at any time be non-functional and we wouldn't know. If we keep it around, what's the point if we don't maintain it right?

**Comment by pospi** [comment]
Based on the fields present in the GraphQL it does seem like something we should partially keep.

@fosterlynn the only updateable `EconomicEvent` properties are `note`, `agreedIn` and `triggeredBy`. Would that be considered "safe" to change from a record keeping perspective? If not, what should be non-updateable?

I think it's probably best to remove `deleteEconomicEvent` entirely given what Lynn is saying above about recordkeeping integrity. Which, really, should be the #1 goal of a project like this. And if we remove that, we should remove `deleteEconomicResource` as well.

`updateEconomicResource`, as mentioned, can stay for now. But there are no tests for this either, and there should be.

**Comment by Connoropolous** [comment]
Ok so I am hearing about some proposals to do some upstream changes in `vf-graphql`. 
Remove
- deleteEconomicResource
- deleteEconomicEvent

having done that, we could remove it here from this hREA project and implementation
